docs/odh/llama-stack-with-odh.md

```diff
@@ -0,0 +1,171 @@
+   - **Additional serving runtime arguments:**
+     ```
+     --dtype=half
+     --max-model-len=20000
+     --gpu-memory-utilization=0.95
+     --enable-chunked-prefill
+     --enable-auto-tool-choice
+     --tool-call-parser=llama3_json
+     --chat-template=/app/data/template/tool_chat_template_llama3.2_json.jinja
+     ```
+
+3. Verify the model deployment:
+```bash
+oc get inferenceservice -n llamastack
+oc get pods -n llamastack | grep llama
+```
+
+### 3. Create LlamaStackDistribution CR
+
+```yaml
+apiVersion: llamastack.io/v1alpha1
+kind: LlamaStackDistribution
+metadata:
+  name: llamastack-custom-distribution
+  namespace: llamastack
+spec:
+  replicas: 1
+  server:
+    containerSpec:
+      env:
+        - name: INFERENCE_MODEL
+          value: llama-32-3b-instruct
+        - name: VLLM_URL
+          value: 'http://llama-32-3b-instruct-predictor.llamastack.svc.cluster.local:80/v1'
+        - name: MILVUS_DB_PATH
+          value: /.llama/distributions/remote-vllm/milvus.db
+      name: llama-stack
+      port: 8321
+    distribution:
+      image: 'quay.io/mcampbel/llama-stack:milvus-granite-embedding-125m-english'
+      podOverrides:
+        volumeMounts:
+          - mountPath: /root/.llama
+            name: llama-storage
+        volumes:
+          - emptyDir: {}
+            name: llama-storage
+```
+
+### 4. Verify LlamaStack Deployment
+
+Check the status of the LlamaStackDistribution:
+```bash
+oc get llamastackdistribution -n llamastack
+```
+
+Check the running pods:
+```bash
+oc get pods -n llamastack | grep llamastack-custom-distribution
+```
+
+Check the logs of the LlamaStack pod:
+```bash
+oc logs -n llamastack -l app=llama-stack
+```
+
+Expected log output should include:
+```
+INFO: Started server process
+INFO: Waiting for application startup.
+INFO: Application startup complete.
+INFO: Uvicorn running on http://['::', '0.0.0.0']:8321
+```
+
+### 5. Query the Model from Jupyter Notebook
+
+1. Go to ODH dashboard -> Workbenches -> Workbench -> Minimal Python
+
+2. Install required packages:
+```python
+%pip install llama_stack llama-stack-client
+```
+
+3. Initialize the client:
+```python
+from llama_stack_client import Agent, AgentEventLogger, RAGDocument, LlamaStackClient
+client = LlamaStackClient(base_url="http://llamastack-custom-distribution-service.llamastack.svc.cluster.local:8321")
+```
+
+4. Verify model registration:
+```python
+client.models.list()
+```
+
+Expected output should show both the LLamA model and the embedding model:
+```
+[Model(identifier='llama-32-3b-instruct', metadata={}, api_model_type='llm',
+       provider_id='vllm-inference', provider_resource_id='llama-32-3b-instruct',
+       type='model', model_type='llm'),
+ Model(identifier='all-MiniLM-L6-v2', metadata={'embedding_dimension': 384.0},
+       api_model_type='embedding', provider_id='sentence-transformers',
+       provider_resource_id='all-MiniLM-L6-v2', type='model', model_type='embedding')]
```

Comment:
We are now using the `ibm-granite` embedding model. I can see this is already adjusted in the `llamastackdistribution` yaml in this PR.
```suggestion
 Model(identifier='ibm-granite/granite-embedding-125m-english', metadata={'embedding_dimension': 768.0},
       api_model_type='embedding', provider_id='sentence-transformers',
       provider_resource_id='ibm-granite/granite-embedding-125m-english', type='model', model_type='embedding')]
```